security-privacy-questionaire.md

```diff
@@ -0,0 +1,71 @@
+
+### 2.6. What information from the underlying platform, e.g. configuration data, is exposed by this specification to an origin?
+
+Anything that exists on disk in files could be exposed by the user to the web. However user agents are encouraged to block list certain directories with particularly sensitive files, and thus somewhat restrict which files and directories the user is allowed to select. For example things like Chrome's "Profile" directory, and other platform configuration data directories are likely going to be on this block list.
+
+### 2.7. Does this specification allow an origin access to sensors on a user’s device
+
+No (unless a device exposes such sensors as "fake" files or directories).
+
+### 2.8. What data does this specification expose to an origin? Please also document what data is identical to data exposed by other features, in the same or different contexts.
+
+The data this specification lets a user expose to an origin is identical to the data exposed via `<input type=file>` and `<input type=file webkitdirectory>`. The differences are in giving the origins the ability to re-prompt for access to files they previously had access to (if handles were stored in IndexedDB), and in the ability for origins to write back to the files (after explicit permission is granted for that).
+
+### 2.9. Does this specification enable new script execution/loading mechanisms?
+
+No.
+
+### 2.10. Does this specification allow an origin to access other devices?
+
+Not really. The exception would be devices that are exposed as "fake" files or directories by the platform. I.e. network shares or cloud storage sync clients could expose data on other devices in a way that looks like regular files or directories. The user agent could let the user pick these files or directories, thereby giving an origin implicit access to this other device. This API doesn't have any functionality to let a website enumerate all network shares on the local network, only explicitly selected files or directories can be accessed by an origin.
+
+### 2.11. Does this specification allow an origin some measure of control over a user agent’s native UI?
+
+Not really. The origin can pop up native file or directory pickers, and have some control over what appears inside that native UI (for example accepted file types), but that control is very limited.
+
+### 2.12. What temporary identifiers might this this specification create or expose to the web?
+
+None.
+
+### 2.13. How does this specification distinguish between behavior in first-party and third-party contexts?
+
+It is expected that user agents do not allow third-party contexts to prompt for any kind of access using this API. I.e. third-party contexts can potentially access files or directories that their origin was already granted access to in a first-party context (by sharing handles via IndexedDB or postMessage), but can't trigger any new file/directory pickers or permission requests.
```

Comment:
I thought we weren't allowing IndexedDB or postMessage acces in a 3P context

Reply:
Our security model doc says "The API will deny requests for user decisions coming from 3rd-party frames.", which to me has always implied that you can still do whatever you want with handles/permissions you already have/somehow got a hold of. Also not sure how meaningful it would be to further restrict things, given that we want service workers to be able to access these handles as well.

Reply:
Got it. Thanks!